test/junit/scala/collection/immutable/LazyListLazinessTest.scala

```diff
@@ -76,13 +76,9 @@ class LazyListLazinessTest {
   }
 
   private def genericFilter_properlyLazy(filter: (LazyList[Int], Int => Boolean) => LazyList[Int],
-                                 isFlipped: Boolean): Unit = {
-    val ops: OpProfileMap = Map(
-      lazyListOp(_.filter(_ => true))  -> NoDrops,
-      lazyListOp(_.filter(i => (i % 2 != 0) == isFlipped)) -> DropProfile(dropCount = 1, repeatedDrops = true),
-    )
-
-    for (op -> d <- ops) genericLazyOp_properlyLazy(op, d)
+                                         isFlipped: Boolean): Unit = {
+    genericLazyOp_properlyLazy(filter(_, _ => !isFlipped))
+    genericLazyOp_properlyLazy(filter(_, i => (i % 2 != 0) == isFlipped), DropProfile(dropCount = 1, repeatedDrops = true))
```

Comment:
all the ones which used to have `OpProfileMap` are now simpler than previously, fewer lines of code, and have the added bonus that the stack trace will tell you which of the two tests (calls?) failed (rather than before, where you can't tell which entry in the `Map` it was.

for this one specifically as well, it was not actually testing the filter function (eta-expanded method) passed in, but rather always testing the `filter` method. also as a consequence of that, the first of the two tests was wrong, but you never noticed because it always tested a method where `isFlipped` was false, so `!isFlipped` was the same as `true`.

Reply:
Good catch!